src/MongoDB.Bson/Serialization/Serializers/ElementAppendingSerializer.cs

```diff
@@ -65,8 +71,7 @@ object IBsonSerializer.Deserialize(BsonDeserializationContext context, BsonDeser
         public void Serialize(BsonSerializationContext context, BsonSerializationArgs args, TDocument value)
         {
             var writer = context.Writer;
-            Action<BsonWriterSettings> settingsConfigurator = s => s.GuidRepresentation = GuidRepresentation.Unspecified;
```

Comment:
This was a hardcoded configurator that applied to all instances of this class. But this isn't right when appending type 1 payloads to a command for OP_QUERY, so now the configurator is passed in to the constructor.